### ISSUE TITLE
parseErrorData must be provided...

### ISSUE THREAD

**Issue by mwawrusch** [magiclabs/magic-flutter#45]
Hi, we upgraded to latest flutter and magic and ran into the following issue:

```
../../../../../.pub-cache/hosted/pub.dev/agent_dart-1.0.0-dev.17/lib/bridge/ffi/ffi_bridge.dart:461:57: Error:
Required named parameter 'parseErrorData' must be provided.
```

This comes from agent_dart which is referenced in the magic sdk. I tried to pinpoint the agent_dart version and this error shows up for the first time in '1.0.0-dev.15'

agent_dart is v 1.0.0-dev.17
magic is 4.1.3

[✓] Flutter (Channel stable, 3.13.7, on macOS 13.5.1 22G90 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.83.0)
[✓] Connected device (3 available)
[✓] Network resources

**Comment by mwawrusch** [comment]
FYI the agent_dart maintainers are working on this,

**Comment by Ariflo** [comment]
Hello @mwawrusch, thanks for bringing this error to our attention. We ran our demo app with the latest version of Flutter and the Magic SDK, keeping `agent_dart` at the `1.0.0-dev.15` version and didn't see the error arise. That said, could you provide more details on the steps you took to produce the error? Does the error occur on app initialization? Have you tried troubleshooting with our demo app?

**Comment by sujithsureshkumar** [comment]
I am also getting the same error.

**Comment by sujithsureshkumar** [comment]
<img width="702" alt="Screenshot 2023-10-27 at 7 20 21 PM" src="https://github.com/magiclabs/magic-flutter/assets/56555135/cb4aea42-b6f4-4cbd-b653-1fb125cbe953">

**Comment by Ariflo** [comment]
@sujithsureshkumar @mwawrusch, Could either of you please provide more information on how you were able to reproduce the error? We are still having trouble reproducing it when running [the demo app](https://github.com/magiclabs/magic-flutter/tree/main/magic_demo) on our end. FWIW, we are running the app on the latest Xcode 15.0.1 / iPhone 15 Pro Max (Simulated) / `magic_sdk: ^4.1.3` / `agent_dart 1.0.0-dev.15`.

**Comment by sujithsureshkumar** [comment]
@Ariflo I got this error when I created a new flutter project using magic sdk. Old project using magic sdk worked fine, but when I removed the magic sdk package from my old project and after that I added again the magic sdk,I got this error in old project as well.

**Comment by Ariflo** [comment]
>@Ariflo I got this error when I created a new flutter project using magic sdk. Old project using magic sdk worked fine, but when I removed the magic sdk package from my old project and after that I added again the magic sdk,I got this error in old project as well.

@sujithsureshkumar Can you clarify what you mean by "removed the  magic sdk package"? Did you remove it from the `pubspec.yaml`, run `flutter run`, then return it to the `pubspec.yaml`? If so, you might need to `flutter clean` to resolve your error. 

Otherwise, I encourage you to troubleshoot with our demo app, as we are still unable to reproduce the error on our end.

**Comment by sujithsureshkumar** [comment]
@Ariflo I had run flutter clean after removing magic sdk from pubspec.yaml. You can reproduce the error by creating new flutter project using magic sdk.

**Comment by Ariflo** [comment]
@sujithsureshkumar thanks for clarifying the steps to reproducing the error. We are now able to see the error on our end and are working on a resolution. Thanks!

**Comment by Ariflo** [comment]
@sujithsureshkumar @mwawrusch Thank you for your patience here. This error took some time to root out as `agent_dart` is  [a nested dependency](https://github.com/AstroxNetwork/flutter_secp256r1/blob/main/pubspec.yaml#L14) of one of our package dependencies used for DPoP. 

That said `agent_dart` maintainers have been diligently working to resolve [the issue](https://github.com/AstroxNetwork/agent_dart/issues/71) we're all seeing with running the package on iOS and will hopefully release a new version with the fix soon. 

With said update in place, we'll push out the changes to our package that should hopefully resolve this issue for everyone.

**Comment by Ariflo** [comment]
@sujithsureshkumar @mwawrusch the `agent_dart` issue has been resolved and fix shipped 🚀 in the latest version of `magic_sdk` v5.0.0. Thanks again for your patience. Happy coding 👋
